bika/lims/browser/analysisrequest/manage_analyses.py

```diff
@@ -179,104 +219,100 @@ def folderitems(self):
             row_data = {}
-            # keyword = obj.getKeyword()
-            # if keyword in review_states.keys() \
-            #    and review_states[keyword] not in ['sample_due',
-            #                                       'to_be_sampled',
-            #                                       'to_be_preserved',
-            #                                       'sample_received',
-            #                                       ]:
-            #     row_data['disabled'] = True
-            items[x]['row_data'] = json.dumps(row_data)
+            item['row_data'] = json.dumps(row_data)
 
             calculation = obj.getCalculation()
-            items[x]['Calculation'] = calculation and calculation.Title()
+            item['Calculation'] = calculation and calculation.Title()
 
             locale = locales.getLocale('en')
             currency = self.context.bika_setup.getCurrency()
             symbol = locale.numbers.currencies[currency].symbol
-            items[x]['before']['Price'] = symbol
-            items[x]['Price'] = obj.getPrice()
-            items[x]['class']['Price'] = 'nowrap'
+            item['before']['Price'] = symbol
+            item['Price'] = obj.getPrice()
+            item['class']['Price'] = 'nowrap'
+            item['Priority'] = ''
 
-            if items[x]['selected']:
-                items[x]['allow_edit'] = ['Partition', 'min', 'max', 'error']
+            if item['selected']:
+                item['allow_edit'] = ['Partition', 'min', 'max', 'error']
                 if not logged_in_client(self.context):
-                    items[x]['allow_edit'].append('Price')
+                    item['allow_edit'].append('Price')
 
-            items[x]['required'].append('Partition')
-            items[x]['choices']['Partition'] = partitions
+            item['required'].append('Partition')
+            item['choices']['Partition'] = partitions
 
             if obj.UID() in self.analyses:
                 analysis = self.analyses[obj.UID()]
                 part = analysis.getSamplePartition()
                 part = part and part or obj
-                items[x]['Partition'] = part.Title()
+                item['Partition'] = part.Title()
                 spec = self.get_spec_from_ar(self.context,
-                                             analysis.getKeyword())
-                items[x]["min"] = spec.get("min",'')
-                items[x]["max"] = spec.get("max",'')
-                items[x]["error"] = spec.get("error",'')
-                items[x]['Price'] = analysis.getPrice()
+                                             analysis.getService().getKeyword())
+                item["min"] = spec.get("min", '')
+                item["max"] = spec.get("max", '')
+                item["error"] = spec.get("error", '')
+                # Add priority premium
+                item['Price'] = analysis.getPrice()
+                item['Priority'] = analysis.getPriority()
```

Comment:
Yes, agree with your appreciation @rockfruit. Anyhow, I think that since this is the "manage_analyses" view from inside AR and the priority assigned to an Analysis corresponds tot that one assigned to the AR to which the analysis belongs, I don't think "Priority" column here being required.